src/libs/antares/study/parts/hydro/series.cpp

```diff
@@ -193,12 +199,108 @@ void DataSeriesHydro::checkMinGenTsNumber(Study& study, const AreaName& areaID)
     }
 }
 
+bool DataSeriesHydro::LoadHydroPowerCredits(Study& study, const AreaName& areaID, const AnyString& folder)
+{
+    bool ret = true;
+    auto& buffer = study.bufferLoadingTS;
+
+    if (study.header.version >= 870)
+    {
+        buffer.clear() << folder << SEP << areaID << SEP << "maxgen." << study.inputExtension;
+        ret = maxgen.loadFromCSVFile(buffer, 1, HOURS_PER_YEAR, &study.dataBuffer) && ret;
+        buffer.clear() << folder << SEP << areaID << SEP << "maxpump." << study.inputExtension;
+        ret = maxpump.loadFromCSVFile(buffer, 1, HOURS_PER_YEAR, &study.dataBuffer) && ret;
+    }
+
+    countpowercredits = maxgen.width;
+
+    if (maxpump.width > countpowercredits)
+        countpowercredits = maxpump.width;
+
+    if (study.usedByTheSolver)
+    {
+        if (countpowercredits == 0)
+        {
+            logs.error() << "Hydro Power Credits: `" << areaID
+                         << "`: empty matrix detected. Fixing it with default values";
+
+            maxgen.reset(1, HOURS_PER_YEAR);
+            maxpump.reset(1, HOURS_PER_YEAR);
+        }
+        else
+        {
+            if (countpowercredits > 1 && maxgen.width != maxpump.width)
+            {
+                if (maxpump.width != 1 && maxgen.width != 1)
+                {
+                    logs.fatal() << "Hydro Power Credits: `" << areaID
+                                 << "`: The matrices Maximum Generation and Maximum Pumping must "
+                                    "have the same number of time-series.";
+                    study.gotFatalError = true;
+                }
+                else
+                {
+                    if (maxpump.width == 1)
+                    {
+                        maxpump.resizeWithoutDataLost(countpowercredits, maxpump.height);
+                        for (uint x = 1; x < countpowercredits; ++x)
+                            maxpump.pasteToColumn(x, maxpump[0]);
```

Comment:
This block is duplicated once.
We should find a solution (like a method / function) to remove the code duplication.